src/cli/vector_generate.rs

```diff
@@ -0,0 +1,279 @@
+            Vec::new(),
+        ));
+    }
+
+    let key = match read_openai_api_key(engine_state) {
+        Ok(k) => k,
+        Err(e) => {
+            return Err(e);
+        }
+    };
+
+    let chunk_len = match call.get_flag::<usize>(engine_state, stack, "chunk")? {
+        Some(l) => l,
+        None => 1024,
+    };
+
+    let dim = match call.get_flag::<i64>(engine_state, stack, "dimension")? {
+        Some(d) => u32::try_from(d).ok().unwrap(),
+        None => 128,
+    };
+
+    let max_tokens: usize = match call.get_flag::<usize>(engine_state, stack, "maxTokens")? {
+        Some(t) => t,
+        None => MAX_FREE_TIER_TOKENS,
+    };
+
+    let mut chunks = Vec::new();
+    let mut batches: Vec<Vec<String>> = Vec::new();
+    let mut results: Vec<Value> = Vec::new();
+
+    if !input.is_none() {
+        chunks.push(input.unwrap());
+        batches.push(chunks.to_vec());
+    } else if !files.is_none() {
+        for file in files.unwrap() {
+            let contents = match fs::read_to_string(file.clone()) {
+                Ok(c) => c,
+                Err(e) => {
+                    return Err(ShellError::GenericError(
+                        format!("Error parsing file {}: {}", file, e),
+                        "".to_string(),
+                        None,
+                        None,
+                        Vec::new(),
+                    ))
+                }
+            };
+
+            let mut iter = contents.chars();
+            let mut pos = 0;
+            while pos < contents.len() {
+                let mut len = 0;
+                for ch in iter.by_ref().take(chunk_len) {
+                    len += ch.len_utf8();
+                }
+                let chunk = &contents[pos..pos + len];
+                chunks.push(chunk.to_string());
+                pos += len;
+            }
+        }
+
+        let bpe = p50k_base().unwrap();
+        let tokens = bpe.encode_with_special_tokens(&chunks.join(" "));
+
+        debug!("Total tokens: {:?}\n", tokens.len());
+
+        let num_batches = (tokens.len() / max_tokens) + 1;
+        let batch_size = chunks.len() / num_batches;
+        if num_batches == 1 {
+            batches.push(chunks.to_vec());
+        } else {
+            let mut lower = 0;
+            let mut upper = batch_size;
+            while lower < chunks.len() {
+                batches.push(chunks[lower..=upper].to_vec());
+                lower = upper + 1;
+                upper += batch_size;
+
+                if upper >= chunks.len() {
+                    upper = chunks.len() - 1;
+                }
+            }
+        };
+    }
+
+    let client =
+        Client::with_config(async_openai::config::OpenAIConfig::default().with_api_key(key));
+
+    let mut vector_count = 0;
+    let start = SystemTime::now();
+    for (i, batch) in batches.iter().enumerate() {
+        let batch_start = SystemTime::now();
+        print!("\rEmbedding batch {:?}/{} ", i + 1, batches.len());
```

Comment:
Should this maybe be info! or debug!

Reply:
Yeah should have been an info!, have made the change.